src/consumer/consumer.js

```diff
@@ -480,6 +483,30 @@ export default class Consumer {
     });
   }
 
+  /**
+   * Adds workspace array to package.json - only if user wants to work with yarn workspaces
+   */
+  async addWorkspacesToPackageJson() {
+    const driver = await this.driver.getDriver();
+    const PackageJson = driver.PackageJson;
+    const pkg = await PackageJson.load(this.getPath(), false);
+    if (pkg) {
+      const workSpaces = pkg.workspaces || [];
+      workSpaces.push(this.bitJson.dependenciesDirectory + BIT_DEPENDECIES_REGEX);
+      const formatedComponentsPath = format(this.bitJson.componentsDefaultDirectory, {
+        name: YARN_WORKSPACES_REGEX,
+        scope: YARN_WORKSPACES_REGEX,
+        namespace: YARN_WORKSPACES_REGEX
+      });
+      const formatedRegexPath = formatedComponentsPath
+        .split(DEFAULT_SEPARATOR)
+        .map(part => (R.contains(YARN_WORKSPACES_REGEX, part) ? YARN_WORKSPACES_REGEX : part))
+        .join(DEFAULT_SEPARATOR);
+      workSpaces.push(formatedRegexPath);
+      pkg.workspaces = R.uniq(workSpaces);
+      await pkg.write({ override: true });
```

Comment:
you override the existing package.json, what will happen to fields that the user defined but are not supported by our packageJson class in bit-javascript? we should make sure we won't delete anything for the user..

Reply:
Amit, regarding Gilad's comment, I had this exact concern when I implemented another task, so in order to keep the root package.json of the user with all the attributes, I didn't use the load() and write() methods of PackageJson.
Instead, I wrote a static method addComponentsIntoExistingPackageJson in the same PackageJson class in bit-javascript, which loads the package.json content raw, add an attribute and save it without creating an instance of PackageJson class.
Feel free to re-use it or write a new similar one.